### PR TITLE
Remove JVM from DateUtils

### PR DESCRIPTION
**Pull request by sgallese**
**PR IN PROGRESS** - Showing migration work in case any obvious issues

For #1075

Starts to remove JVM dependencies from DateUtils, migrate to common LocalDate

Depends on https://github.com/iSoron/uhabits/pull/1111 (will rebase on dev once merged)

Next up

- [ ] Finish tests for Timestamp.kt
- [ ] Move Timestamp/Test.kt from jvm to core
- [ ] Remove fluid-locale

### CODE REVIEW

**Comment by sgallese** [comment]
Odd gradle error on CI test failure.
I was able to run `:uhabits-core:allTests` task locally without issue.

**Comment by sgallese** [comment]
I need to investigate test failure!

File: uhabits-core/build.gradle.kts [hunk 1]
```diff
@@ -30,6 +30,8 @@ kotlin {
             dependencies {
                 implementation(kotlin("stdlib-common"))
                 implementation("org.jetbrains.kotlinx:kotlinx-coroutines-core-common:1.3.8")
+                implementation("org.jetbrains.kotlinx:kotlinx-datetime:0.2.1")
+                implementation("io.fluidsonic.locale:fluid-locale:0.10.0")
```
Comment (sgallese) [inline uhabits-core/build.gradle.kts:34]: @iSoron OK to use this library for Locale? Or do we need to role our own per platform?

Comment (iSoron) [reply uhabits-core/build.gradle.kts:34]: It looks like the library's main functionality is parsing language tags (e.g. "ZH-HANT-cn-somevar"). Unless I am missing something, I don't think we need this functionality; we can just store the tag as is. For example, it seems to me like the following `Locale` class would suffice for our purposes, but please let me know if I'm mistaken:

```kotlin
data class Locale(val tag: String)
```

**Comment by hiqua** [comment]
@sgallese are you still working on this?

**Comment by sgallese** [comment]
Unfortunately this task seems to continually grow :( I'm thinking I should
abandon in favor of smaller, quicker wins on removing the JVM. Ok to close
this PR.

**Pull request closed without merging**